R/sugar.R

```diff
@@ -20,3 +20,162 @@ tnr = function(.key, ...) {
 tnrs = function(.keys, ...) {
   dictionary_sugar_mget(mlr_tuners, .keys, ...)
 }
+
+#' @title Syntactic Sugar for Automatic Tuning
+#' 
+#' @description
+#' Function to create an [AutoTuner].
+#' 
+#' @param method (`character(1)`)\cr
+#'  Key to retrieve tuner from [mlr_tuners] dictionary.
+#' @param term_evals (`integer(1)`)\cr
+#'  Number of allowed evaluations.
+#' @param term_time (`integer(1)`)\cr
+#'  Maximum allowed time in seconds.
+#' @param ... (named `list()`)\cr
+#'  Named arguments to be set as parameters of the tuner.
+#' 
+#' @return [AutoTuner]
+#' 
+#' @template param_learner
+#' @template param_resampling
+#' @template param_measure
+#' @template param_search_space
+#' 
+#' @export
+#' @examples
+#' learner = lrn("classif.rpart")
+#' learner$param_set$values$minsplit = to_tune(1, 10)
+#'
+#' at = tune_auto(
+#'   method = "random_search",
+#'   learner = learner, 
+#'   resampling = rsmp ("holdout"),
+#'   measure = msr("classif.ce"), 
+#'   term_evals = 50, 
+#'   batch_size = 10)  
+#'
+#' at$train(tsk("pima"))
+tune_auto = function(method, learner, resampling, measure, term_evals = NULL, term_time = NULL, search_space = NULL,
+  ...) {
+  assert_choice(method, mlr_tuners$keys())
+  assert_int(term_evals, null.ok = TRUE)
+  assert_int(term_time, null.ok = TRUE)
+
+  terminator = if (is.null(term_evals) && is.null(term_time)) {
+    stop("`term_evals` or `term_time` must be provided")
+  } else if (!is.null(term_evals) && !is.null(term_time)) {
+    trm("combo", list(trm("evals", n_evals = term_evals), trm("run_time", secs = term_time)))
+  } else if (!is.null(term_evals)) {
+    trm("evals", n_evals = term_evals)
+  } else if (!is.null(term_time)) {
+    trm("run_time", secs = term_time)
+  }
+
+  tuner = tnr(method, ...)
+  AutoTuner$new(learner, resampling, measure, terminator, tuner, search_space)
+}
+
+#' @title Syntactic Sugar for Tuning
```

Comment:
```suggestion
#' @title Function for Tuning
```
It's a bit more than sugar?